buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/gradlebuild/unittestandcompile/UnitTestAndCompilePlugin.kt

```diff
@@ -78,23 +78,22 @@ class UnitTestAndCompilePlugin : Plugin<Project> {
 
             tasks.withType<JavaCompile>().configureEach {
                 options.isIncremental = true
-                configureCompileTask(this, options, availableJavaInstallations, extension.moduleType.requiredCompiler)
+                configureCompileTask(this, options, availableJavaInstallations)
             }
             tasks.withType<GroovyCompile>().configureEach {
                 groovyOptions.encoding = "utf-8"
-                configureCompileTask(this, options, availableJavaInstallations, extension.moduleType.requiredCompiler)
+                configureCompileTask(this, options, availableJavaInstallations)
             }
         }
         addCompileAllTask()
     }
 
     private
-    fun configureCompileTask(compileTask: AbstractCompile, options: CompileOptions, availableJavaInstallations: AvailableJavaInstallations, requiredCompilerVersion: JavaVersion) {
+    fun configureCompileTask(compileTask: AbstractCompile, options: CompileOptions, availableJavaInstallations: AvailableJavaInstallations) {
         options.isFork = true
         options.encoding = "utf-8"
         options.compilerArgs = mutableListOf("-Xlint:-options", "-Xlint:-path")
-        val compilerJdkVersion = maxOf(requiredCompilerVersion, JavaVersion.VERSION_1_7)
-        val jdkForCompilation = availableJavaInstallations.jdkForCompilation(compilerJdkVersion)
+        val jdkForCompilation = availableJavaInstallations.javaInstallationForCompilation
```

Comment:
We could also set the `--release` command line option when the targetCompatibility for the project is >= 8. This is not very interesting right now I guess, since we don't have many projects where this is the case.